src/ConstructiveSolidGeometry/SurfacePrimitives/Plane.jl

```diff
@@ -1,168 +1,51 @@
-struct Plane{T,TP,TA} <: AbstractSurfacePrimitive{T}
-    p1::TP
-    p2::TP
-    p3::TP
-    p4::TA
-    function Plane( ::Type{T},
-                   p1::CartesianPoint{T},
-                   p2::CartesianPoint{T},
-                   p3::CartesianPoint{T},
-                   p4:: Union{Nothing, CartesianPoint{T}}) where {T}
-        new{T, CartesianPoint{T}, typeof(p4)}(p1, p2, p3, p4)
-    end
+struct Plane{T} <: AbstractFlatSurfacePrimitive{T}
```

Comment:
I would opt for `Planar` and `Curved` instead of `Flat` and `Bent` Abstract...SurfacePrimitive

Reply:
I changed that: https://github.com/JuliaPhysics/SolidStateDetectors.jl/pull/163/commits/61984e0f1f13bc2205d2b421c49ecf91b37f6bcf